### PR TITLE
[PW-2250] Outsource the builders from PrestaShop to the php api library

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**
Outsource the builders from PrestaShop to the php api library

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: tests/Integration/Builder/OpenInvoiceTest.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use Adyen\Service\Builder\OpenInvoice;
+use Adyen\TestCase;
+
+class OpenInvoiceTest extends TestCase
+{
+    public function testBuildOpenInvoiceLineItem()
+    {
+        $expectedResult = array(
+            'id' => 1,
+            'itemId' => 1,
```
Comment (Aleffio) [inline tests/Integration/Builder/OpenInvoiceTest.php:13]: id is a string: https://docs.adyen.com/payment-methods/klarna/invoice-lines#payment-request

File: src/Adyen/Util/OpenInvoice.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * For Klarna And AfterPay use Vat category High others use none
+     *
+     * @param $paymentMethod
+     * @return string 'High'/'None'
+     */
+    public function getVatCategory($paymentMethod)
+    {
+        if ($paymentMethod == "klarna" ||
+            strlen($paymentMethod) >= 9 &&
+            mb_substr($paymentMethod, 0, 9) == 'afterpay_'
```
Comment (cyattilakiss) [inline src/Adyen/Util/OpenInvoice.php:50]: I'm not really sure how it was working before but in this scenario only klarna payment and afterpay_b2b, afterpay_deafult and afterpay_directdebit is taken into consideration so klarna_* and afterpay is not. So I propose the following:


```suggestion
        if (mb_substr($paymentMethod, 0, 6) == self::KLARNA_PAYMENT_METHOD ||
            mb_substr($paymentMethod, 0, 8) == self::AFTERPAY_PAYMENT_METHOD
```

File: tests/Integration/Builder/PaymentTest.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                'allow3DS2' => true
+            ),
+            'channel'=>'web',
+            'origin'=>'theorigin'
+        );
+        $payment = new Payment();
+        $result = $payment->build3DS2Data('theorigin', array());
+        $this->assertEquals($result, $expectedResult);
+    }
+}
```
Comment (cyattilakiss) [inline tests/Integration/Builder/PaymentTest.php:72]:
```suggestion
}

```

File: src/Adyen/Service/Builder/Browser.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        if (!empty($language)) {
+            $request[$browserInfo]['language'] = $language;
+        }
+
+        $request[$browserInfo]['javaEnabled'] = $javaEnabled;
+
+        return $request;
+    }
+}
```
Comment (cyattilakiss) [inline src/Adyen/Service/Builder/Browser.php:85]:
```suggestion
}

```

File: tests/Integration/Builder/RefundTest.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        $result = $refund->buildRefundRequest(
+            100,
+            "12345",
+            "812345679",
+            "EUR",
+            "TestMerchantAccount"
+        );
+        $this->assertEquals($result, $expectedResult);
+    }
+}
```
Comment (cyattilakiss) [inline tests/Integration/Builder/RefundTest.php:53]:
```suggestion
}

```

File: tests/Integration/Builder/OpenInvoiceTest.php (hunk 1 above)
Comment (Aleffio) [reply tests/Integration/Builder/OpenInvoiceTest.php:35]: please also change this and the phpdoc: https://github.com/Adyen/adyen-php-api-library/blob/d4d2a27046a8b48c367b6142b4c532b9b71a2d67/src/Adyen/Service/Builder/OpenInvoice.php#L47

File: src/Adyen/Util/OpenInvoice.php [hunk 6]
```diff
@@ -0,0 +1,84 @@
+<?php
```
Comment (Aleffio) [inline src/Adyen/Util/OpenInvoice.php:1]: copyright please

File: tests/Integration/Builder/CustomerTest.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            'Smith',
+            'NL',
+            '',
+            '',
+            '',
+            $request
+        );
+        $this->assertEquals($result, $expectedResult);
+    }
+}
```
Comment (Aleffio) [inline tests/Integration/Builder/CustomerTest.php:64]:
```suggestion
}

```

File: tests/Unit/Util/OpenInvoiceTest.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (Aleffio) [inline tests/Unit/Util/OpenInvoiceTest.php:1]: copyright please

File: src/Adyen/Service/Builder/Browser.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        $acceptHeader = '',
+        $screenWidth = 0,
+        $screenHeight = 0,
+        $colorDepth = 0,
+        $timeZoneOffset = 0,
+        $language = '',
+        $javaEnabled = false,
+        $request = array()
+    ) {
+        $browserInfo = 'browserInfo';
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply src/Adyen/Service/Builder/Browser.php:52]: Sonar cloud doesn't like private const and I added in the method body .

**Pull request merged**